testfixtures/shouldraise.py

```diff
@@ -89,3 +90,20 @@ def _should_raise_wrapper(*args, **kw):
                 target(*args, **kw)
 
         return _should_raise_wrapper
+
+
+@contextmanager
+def ShouldAssert(expected_text):
+    """
+    A context manager to check that an :class:`AssertionError`
+    is raised and its text is as expected.
+    """
+    try:
+        yield
+    except AssertionError as e:
+        actual_text = str(e)
+        if expected_text != actual_text:
+            raise AssertionError(diff(expected_text, actual_text,
+                                      x_label='expected', y_label='actual'))
+    else:
+        raise AssertionError('No exception raised!')
```

Comment:
The failure case for `with ShouldAssert(...)` is going to provide less meaningful information "No exception raised!", as compared to `with ShouldRaise(AssertionError(...))` which will fail with a message that mentions what was expected.

I think it would be helpful to include the `expected_text` here.

Reply:
Done.